pxf make error (make test & make -C)

Hi,

I have an centos7 env with gpdb 6.23.0 running on it. 
Then I cloned the pxf code into the contrib folder under gpdb source code path. 
when I was trying to build pxf, I came across these errors:

$ make 
BUILD SUCCESSFUL

$ make test
[1674715619] Cmd Suite - 67/67 specs ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• SUCCESS! 5.04176ms PASS
[1674715619] EndToEnd Suite - 5/5 specs ••••• SUCCESS! 16.486254ms PASS

Ginkgo ran 2 suites in 4.858285575s
Test Suite Passed
make[1]: Leaving directory `/home/gpadmin/workspace/pxf/cli/go/src/pxf-cli'
make -C server test
make[1]: Entering directory `/home/gpadmin/workspace/pxf/server'
./gradlew -Pversion=6.5.1-SNAPSHOT -PapiVersion=16 test

> Task :pxf-service:test

ProfilesConfTest > testMalformedXmlFile() FAILED
    org.opentest4j.AssertionFailedError at ProfilesConfTest.java:196

248 tests completed, 1 failed, 1 skipped

> Task :pxf-service:test FAILED

FAILURE: Build failed with an exception.

So I modified the java source "ProfilesConfTest.java" as below :

[gpadmin@mdw profile]$ pwd
/home/gpadmin/workspace/pxf-release-6.3.1/server/pxf-service/src/test/java/org/greenplum/pxf/service/profile
-----------
/*
    @Test
    public void testMalformedXmlFile() {
        Exception e = assertThrows(ProfileConfException.class,
                () -> getProfilesConf("malformedXmlFile"));
        assertTrue(e.getMessage().contains("Content is not allowed in prolog"));
    }
*/
-----------

And then build succeeded.
I have 2 question.

1) Is there any problem if I modify the source as above and use it?
2) If that doesn't work, can you fix that error?
ProfilesConfTest > testMalformedXmlFile() FAILED
    org.opentest4j.AssertionFailedError at ProfilesConfTest.java:196

248 tests completed, 1 failed, 1 skipped

Best Regards
Kai

--- a/server/pxf-service/src/test/java/org/greenplum/pxf/service/profile/ProfilesConfTest.java
+++ b/server/pxf-service/src/test/java/org/greenplum/pxf/service/profile/ProfilesConfTest.java
@@ -23,6 +23,8 @@ import org.junit.jupiter.api.Test;
 
 import java.util.Map;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.matchesPattern;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
@@ -193,7 +195,7 @@ public class ProfilesConfTest {
     public void testMalformedXmlFile() {
         Exception e = assertThrows(ProfileConfException.class,
                 () -> getProfilesConf("malformedXmlFile"));
-        assertTrue(e.getMessage().contains("Content is not allowed in prolog"));
+        assertThat(e.getMessage(), matchesPattern("^Profiles configuration .+ could not be loaded:.*"));
     }
 
     @Test

